event-loop.c

```diff
@@ -52,6 +114,36 @@ int kanshi_main_loop(struct kanshi_state *state) {
 			return EXIT_FAILURE;
 		}
 
+		if (readfds[FD_SIGNAL].revents & POLLIN) {
+			for (;;) {
+				int signum;
+				ssize_t s
+					= read(readfds[FD_SIGNAL].fd, &signum, sizeof(signum));
+				if (s == 0) {
+					break;
+				}
+				if (s < 0) {
+					if (errno == EAGAIN) {
+						break;
+					}
+					fprintf(stderr, "read from signal pipe failed: %s\n",
+							strerror(errno));
+					return EXIT_FAILURE;
+				}
+				if (s < (ssize_t) sizeof(signum)) {
+					fprintf(stderr, "read too few bytes from signal pipe\n");
+					return EXIT_FAILURE;
+				}
+				switch (signum) {
+				case SIGHUP:
+					kanshi_reload_config(state);
+					break;
+				default:
+					return signum | 0x80;
```

Comment:
Why this magic number?

Reply:
This is from the original PR. I assume it came about from wanting the return value to show the program was killed by a signal. Will confirm it works portably and add a comment, otherwise just remove, I guess?

Reply:
For now, I will consider it a success instead and just exit with 0 on signals...

Reply:
I'd be fine with that.